### PR TITLE
chore: min change

### LINKED ISSUE #4: Update Avrdude fork
Update to the latest upstream version. Also: restore some of the removed functions/files so updating will be easier in the future (like the signature check during upload, etc.)

### PR DESCRIPTION
**Pull request by koen1711**

### CODE REVIEW

File: NEWS
Comment (rmoesbergen) [inline NEWS:1]: I'd leave this to prevent future merge conflicts

File: libserial/LibSerial.c [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+    set_dts_rts(is_on);
+}
+
+void serialPortDrain(int timeout) {
+    readBufferLen = 0;
+    clear_read_buffer(timeout);
+}
+
+void serialPortWrite(const unsigned char *buf, size_t len) {
+//    std::vector<unsigned char> data(buf, buf + len);
```
Comment (rmoesbergen) [inline libserial/LibSerial.c]: remove?

File: libserial/LibSerial.c [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+    write_data((unsigned char*)buf, (int)len);
+}
+
+int serialPortRecv(unsigned char *buf, size_t len, int timeoutMs) {
+    read_data(timeoutMs, len);
+    size_t processed = 0;
+    if (readBufferLen != 0) {
+        // check how much data is needed and add that much to the buffer
+        if (readBufferLen >= len) {
+//            data = std::vector<unsigned char>(readBuffer.begin(), readBuffer.begin() + len);
```
Comment (rmoesbergen) [inline libserial/LibSerial.c]: remove?

**Review by sverben — APPROVED** [review]
Please bump version to fix webbased

**Review by rmoesbergen — APPROVED** [review]

**Pull request merged**